### PR TITLE
Promoted Content: Add and configure new promoted content view.

### PR DESCRIPTION
**Pull request by beckyvb**

### CODE REVIEW

**Comment by beckyvb** [comment]
Relies on https://github.com/vanilla/vanilla/pull/2747/

File: applications/vanilla/views/modules/promoted.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+   $categoryName = val('CategoryName', $row);
+   $date         = val('DateUpdated', $row) ?: val('DateInserted', $row);
+   $date         = Gdn_Format::Date($date, 'html');
+   $type         = val('RecordType', $row, 'post');
+   $id           = val('CommentID', $row, val('DiscussionID', $row, ''));
+   $author       = val('Author', $row);
+   $username     = val('Name', $author);
+   $userUrl      = val('Url', $author);
+   $userPhoto    = val('PhotoUrl', $author);
+   $cssClass     = val('_CssClass', $author);
```
Comment (linc) [inline applications/vanilla/views/modules/promoted.php]: Typically we use these `val`s in context rather than calling a big stack of them like this.

**Comment by linc** [comment]
I feel like these sorts of functions wanna be in a helper_functions type of file typically, no? Overriding this view would mean killing all these functions right now too.

File: applications/vanilla/views/modules/promoted.php (hunk 1 above)
Comment (beckyvb) [inline applications/vanilla/views/modules/promoted.php]: I'm writing my views with the idea that we may eventually decide to use templates for our html views. I'm trying to separate the data collection and logic from the html/view as much as possible to make that conversion as smooth as possible.

**Comment by beckyvb** [comment]
Ready for review again. This is potentially breaking for any theme that has overridden the modules/helper_functions view.

**Pull request merged**